components/gripper/fake/gripper.go

```diff
@@ -66,9 +66,23 @@ func (g *Gripper) Reconfigure(_ context.Context, _ resource.Dependencies, conf r
 	return nil
 }
 
-// ModelFrame returns the dynamic frame of the model.
-func (g *Gripper) ModelFrame() referenceframe.Model {
-	return g.model
+// Kinematics returns the kinematic model associated with the gripper.
+func (g *Gripper) Kinematics(ctx context.Context) (referenceframe.Model, error) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	return g.model, nil
+}
+
+// CurrentInputs is unimplemented for grippers.
+func (g *Gripper) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
```

Comment:
If these are new methods & unimplemented, why not return unimplemented errors?

Reply:
I didn't write these lines.  But yes good idea, will fix implementation, both here and in the client